pkg/web/routes.go

```diff
@@ -672,31 +673,55 @@ func (wa *WebAPI) buildPipelines(p Push, rawPush []byte, f dinghyfile.Downloader
 		return
 	}
 
+	modulesProcessed :=0
 	// Check if we're in a template repo
 	if p.Repo() == settings.TemplateRepo {
 		// Set status to pending while we process modules
 		p.SetCommitStatus(settings.InstanceId, git.StatusPending, git.DefaultMessagesByBuilderAction[builder.Action][git.StatusPending])
-
+		var filesToIgnore []string
+		//check for dinghyignore file
+		patternsToIgnore, err :=f.Download(p.Org(),p.Repo(), ".dinghyignore", p.Branch())
+		if err != nil {
+			dinghyLog.Info(".dinghyignore file not found in template repository, validating all files in the push")
+		} else {
+			dinghyLog.Infof(".dinghyignore file found! Ignoring files that match these globs: %s", patternsToIgnore)
+			filesToIgnore = strings.Split(patternsToIgnore, "\n")
+		}
 		// For each module pushed, rebuild dependent dinghyfiles
 		for _, file := range p.Files() {
-			// ensure module is correctly parsed
-			if _, err := builder.Parser.Parse(p.Org(), p.Repo(), file, p.Branch(), nil); err != nil {
-				p.SetCommitStatus(settings.InstanceId, git.StatusError, "module parse failed")
-				dinghyLog.Errorf("module parse failed: %s", err.Error())
-				saveLogEventError(wa.LogEventsClient, p, dinghyLog, logevents.LogEvent{RawData: string(rawPush), PullRequest: pullRequest, RenderedDinghyfile: dinghyfilesRendered})
-				return
+			var shouldIgnore bool
+			//check to see if the file should be ignored
+			for _ ,pattern := range  filesToIgnore {
+				if pattern != ""{
+					shouldIgnore, _ = regexp.MatchString(pattern, file)
+					if shouldIgnore {
```

Comment:
im wondering if this will escape a regex like `/somefile/`

Reply:
We will make sure the docs reflect that it uses lazy pattern matching and does not auto escape characters. `/somefile/` would not escape the `/` and would escape the s